### PR TITLE
Unicode Fix for serialized objects saved for credentials and added Credential UI Tests . closes #191 closes #179

### LINKED ISSUE #179: Add tests for Credentials UI

### LINKED ISSUE #191: Harvest Job unable to read credentials because of Unicode characters.
Our sfm ui credential forms saves credentials with unicode characters which harvesters are unable to read  as experienced in https://github.com/gwu-libraries/sfm-ui/issues/187

They are saved as:
```
token = {'consumer_secret': u'dummysecret', 'access_token': u'dummytoken', 'consumer_key': u'dummykey', 'access_token_secret': u'dummytokensecret'}
```
as opposed to 
```
token = {"consumer_secret": "dummysecret", "access_token": "dummytoken", "consumer_key": "dummykey", "access_token_secret": "dummytokensecret"}
```
We are getting following error because our job is unable to read unicode characters

```
Traceback (most recent call last):
sfmdevapp_1               |   File "/usr/local/lib/python2.7/dist-packages/apscheduler/executors/base.py", line 108, in run_job
sfmdevapp_1               |     retval = job.func(*job.args, **job.kwargs)
sfmdevapp_1               |   File "/usr/local/lib/python2.7/dist-packages/django/utils/decorators.py", line 145, in inner
sfmdevapp_1               |     return func(*args, **kwargs)
sfmdevapp_1               |   File "/opt/sfm-ui/sfm/ui/jobs.py", line 50, in seedset_harvest
sfmdevapp_1               |     message["credentials"] = json.loads(str(historical_credential.token))
sfmdevapp_1               |   File "/usr/lib/python2.7/json/__init__.py", line 338, in loads
sfmdevapp_1               |     return _default_decoder.decode(s)
sfmdevapp_1               |   File "/usr/lib/python2.7/json/decoder.py", line 366, in decode
sfmdevapp_1               |     obj, end = self.raw_decode(s, idx=_w(s, 0).end())
sfmdevapp_1               |   File "/usr/lib/python2.7/json/decoder.py", line 382, in raw_decode
sfmdevapp_1               |     obj, end = self.scan_once(s, idx)
sfmdevapp_1               | ValueError: Expecting property name: line 1 column 2 (char 1)

```

### PR DESCRIPTION
**Pull request by rajatvij**
This saves credentials in correct json format without unicode characters so harvest job can read credentials.

### CODE REVIEW

**Comment by rajatvij** [comment]
@kerchner This branch can be tested now. This branch encapsulates the changes i did for #179  as the test for token value could only be done after this fix and it was cleaner to include that in the tests i already added in #193.

**Comment by rajatvij** [comment]
This branch contains fixes #191  for Unicode error we were getting while harvesting. This was due to forms saving python serialized objects for credentials fields related to token.
Additionally, This branch adds test cases for credential UI. #179 
This branch encapsulates #193, hence closes #193

File: sfm/ui/test_forms.py [hunk 1]
```diff
@@ -163,3 +164,114 @@ def setUp(self):
...
+                                                    token='{"key1":"key1","key2":"key2"}')
+        self.path = '/ui/credentials/' + str(self.credential.pk) + '/update/'
+
+    def test_valid_data(self):
+        request = self.factory.get(self.path)
+        response = CredentialUpdateView.as_view()(request,
+                                                  pk=self.credential.pk)
+        form = CredentialForm({
+            'name': 'my test credential updated name',
+            'token': '{"key1":"updated_key1","key2","update_key2"}',
```
Comment (lwrubel) [inline sfm/ui/test_forms.py]: Did you intend this to not be json? Not sure it has any effect on the test, but would be clearer.

File: sfm/ui/test_forms.py [hunk 2]
```diff
@@ -163,3 +164,114 @@ def setUp(self):
...
+            "secret":"dummy_secret",
+            "date_added": "04/14/2016",
+        }
+
+    def test__form(self):
+        form = CredentialFlickrForm(self.data)
+        form.instance.user = self.user
+        self.assertTrue(form.is_valid())
+        credential = form.save()
+        self.assertTrue(credential.token,
```
Comment (lwrubel) [inline sfm/ui/test_forms.py]: assertTrue is checking that there is a token and not that it's equal to the string that follows. Use assertEqual instead?

File: sfm/ui/test_forms.py [hunk 3]
```diff
@@ -163,3 +164,114 @@ def setUp(self):
...
+            "access_token_secret": "dummy_access_token_secret",
+            "date_added": "04/14/2016",
+        }
+
+    def test_form(self):
+        form = CredentialTwitterForm(self.data)
+        form.instance.user = self.user
+        self.assertTrue(form.is_valid())
+        credential = form.save()
+        self.assertTrue(credential.token,
```
Comment (lwrubel) [inline sfm/ui/test_forms.py]: assertTrue is checking that there is a token and not that it's equal to the string that follows. Use assertEqual instead?

File: sfm/ui/test_forms.py [hunk 4]
```diff
@@ -163,3 +164,114 @@ def setUp(self):
...
+            "access_token": "dummy_access_token",
+            "date_added": "04/14/2016",
+        }
+
+    def test_form(self):
+        form = CredentialWeiboForm(self.data)
+        form.instance.user = self.user
+        self.assertTrue(form.is_valid())
+        credential = form.save()
+        self.assertTrue(credential.token,
```
Comment (lwrubel) [inline sfm/ui/test_forms.py]: assertTrue is checking that there is a token and not that it's equal to the string that follows. Use assertEqual instead?

File: sfm/ui/test_forms.py [hunk 5]
```diff
@@ -163,3 +164,114 @@ def setUp(self):
...
+        self.data = {
+            "name": "test_flickr_credential",
+            "user": self.user.pk,
+            "platform": "flickr",
+            "key":"dummy_key",
+            "secret":"dummy_secret",
+            "date_added": "04/14/2016",
+        }
+
+    def test__form(self):
```
Comment (lwrubel) [inline sfm/ui/test_forms.py]: typo? should be `test_form` like in other TestCases?

**Comment by lwrubel** [comment]
@rajatvij a few comments, see files changed tab.

File: sfm/ui/test_forms.py (hunk 1 above)
Comment (rajatvij) [inline sfm/ui/test_forms.py]: Yes. As this is the value for token field, which can be any string and not necessarily JSON, although it makes more sense to make it JSON, as it makes it more clearer. So i am changing it as per your comment as it doesn't have any effect on test. We are just checking validity of the form here. Let me if you think if this change was not required.

**Comment by rajatvij** [comment]
@lwrubel I have corrected the test as per your comments. Regarding Credential Update Test I did change the test as per your comment, but please check my response and let me know if you think this change looks good to you?

**Comment by lwrubel** [comment]
thanks @rajatvij, taking a look. Using assertJSONEqual, even better.

**Pull request merged**